langstream-api-gateway/src/main/java/ai/langstream/apigateway/gateways/ConsumeGateway.java

```diff
@@ -0,0 +1,224 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package ai.langstream.apigateway.gateways;
+
+import ai.langstream.api.model.Gateway;
+import ai.langstream.api.model.StreamingCluster;
+import ai.langstream.api.runner.code.Header;
+import ai.langstream.api.runner.code.Record;
+import ai.langstream.api.runner.topics.TopicConnectionsRuntime;
+import ai.langstream.api.runner.topics.TopicConnectionsRuntimeRegistry;
+import ai.langstream.api.runner.topics.TopicOffsetPosition;
+import ai.langstream.api.runner.topics.TopicReadResult;
+import ai.langstream.api.runner.topics.TopicReader;
+import ai.langstream.apigateway.websocket.AuthenticatedGatewayRequestContext;
+import ai.langstream.apigateway.websocket.api.ConsumePushMessage;
+import ai.langstream.apigateway.websocket.api.ProduceResponse;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.Closeable;
+import java.util.Base64;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CancellationException;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Executor;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.function.Supplier;
+import lombok.Getter;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class ConsumeGateway implements Closeable {
+
+    protected static final ObjectMapper mapper = new ObjectMapper();
+
+    @Getter
+    public static class ProduceException extends Exception {
+
+        private final ProduceResponse.Status status;
+
+        public ProduceException(String message, ProduceResponse.Status status) {
+            super(message);
+            this.status = status;
+        }
+    }
+
+    public static class ProduceGatewayRequestValidator
+            implements GatewayRequestHandler.GatewayRequestValidator {
+        @Override
+        public List<String> getAllRequiredParameters(Gateway gateway) {
+            return gateway.getParameters();
+        }
+
+        @Override
+        public void validateOptions(Map<String, String> options) {
+            for (Map.Entry<String, String> option : options.entrySet()) {
+                switch (option.getKey()) {
+                    default -> throw new IllegalArgumentException(
+                            "Unknown option " + option.getKey());
+                }
+            }
+        }
+    }
+
+    private final TopicConnectionsRuntimeRegistry topicConnectionsRuntimeRegistry;
+    private TopicReader reader;
+    private AuthenticatedGatewayRequestContext requestContext;
+    private List<Function<Record, Boolean>> filters;
+
+    public ConsumeGateway(TopicConnectionsRuntimeRegistry topicConnectionsRuntimeRegistry) {
+        this.topicConnectionsRuntimeRegistry = topicConnectionsRuntimeRegistry;
+    }
+
+    @SneakyThrows
```

Comment:
SneakyThrows often hide potential problems, do we really need it ?
what happens if there is an error in anyplace ? we risk to leak resources